core/src/main/scala/org/apache/spark/deploy/master/FileSystemPersistenceEngine.scala

```diff
@@ -58,7 +58,7 @@ private[spark] class FileSystemPersistenceEngine(
     if (!created) { throw new IllegalStateException("Could not create file: " + file) }
     val serializer = serialization.findSerializerFor(value)
     val serialized = serializer.toBinary(value)
-    val out = new FileOutputStream(file)
+    val out = new BufferedOutputStream(new FileOutputStream(file))
```

Comment:
Although the buffering in `ExternalSorter` makes sense, I don't know if this helps anything, since only one write is being done. Another buffer at this level doesn't do anything but introduce a copy and memory allocation